### PR TITLE
Hide journal recovery messages during test suite run

### PR DESCRIPTION
**Pull request by adamretter**
Stops the CI logs from getting polluted

### CODE REVIEW

File: exist-core/src/main/java/org/exist/storage/recovery/RecoveryManager.java [hunk 1]
```diff
@@ -135,6 +139,14 @@ public boolean recover() throws LogException {
                                 lastCheckpoint = (Checkpoint) next;
                             }
                             lastLsn = next.getLsn();
+
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/storage/recovery/RecoveryManager.java:142]: I was thinking of this, great catch!

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -35,4 +35,4 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: mvn -V -B -Dsurefire.useFile=false -DtrimStackTrace=false -Ddependency-check.skip=true -Ddocker=false -P \!mac-dmg-on-mac,\!codesign-mac-dmg,\!mac-dmg-on-unix,\!installer,\!concurrency-stress-tests,\!micro-benchmarks,\!build-dist-archives verify site org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
+        run: mvn -V -B -Dsurefire.useFile=false -DtrimStackTrace=false -Ddependency-check.skip=true -Ddocker=false -P \!mac-dmg-on-mac,\!codesign-mac-dmg,\!mac-dmg-on-unix,\!installer,\!concurrency-stress-tests,\!micro-benchmarks,\!build-dist-archives verify site org.sonarsource.scanner.maven:sonar-maven-plugin:4.0.0.4121:sonar
```
Comment (dizzzz) [inline .github/workflows/sonarcloud.yml:38]: is it an good idea to have this versionnr hardcoded in the scripts?

**Review by dizzzz — COMMENTED** [review]
It looks like the PR contains content of another PR?

File: .github/workflows/ci-deploy.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: liberica
```
Comment (reinhapa) [inline .github/workflows/ci-deploy.yml:16]: How does a different Java implementation impact the journal recovery output?

File: .github/workflows/ci-test.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: liberica
```
Comment (reinhapa) [inline .github/workflows/ci-test.yml:16]: How does a different Java implementation impact the journal recovery output?

File: .github/workflows/ci-test.yml [hunk 5]
```diff
@@ -25,7 +25,7 @@ jobs:
...
```
Comment (reinhapa) [inline .github/workflows/ci-test.yml:28]: How does a different Java implementation impact the journal recovery output?

File: .github/workflows/ci-test.yml [hunk 6]
```diff
@@ -46,14 +46,14 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: liberica
```
Comment (reinhapa) [inline .github/workflows/ci-test.yml:49]: How does a different Java implementation impact the journal recovery output?

File: .github/workflows/ci-xqts.yml [hunk 7]
```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: liberica
```
Comment (reinhapa) [inline .github/workflows/ci-xqts.yml:15]: How does a different Java implementation impact the journal recovery output?

File: .github/workflows/sonarcloud.yml [hunk 8]
```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: liberica
```
Comment (reinhapa) [inline .github/workflows/sonarcloud.yml:20]: How does a different Java implementation impact the journal recovery output?

**Review by reinhapa — APPROVED** [review]
Great enhancement to have less output during tests!

File: .github/workflows/ci-deploy.yml (hunk 3 above)
Comment (dizzzz) [reply .github/workflows/ci-deploy.yml:16]: the PR combines 2 PRs in essence; the change is in the recoverymanager

File: .github/workflows/sonarcloud.yml (hunk 2 above)
Comment (adamretter) [reply .github/workflows/sonarcloud.yml:38]: It was using a really old version before by default. The only way I could see to get it to use the latest was to add it to the command here.

**Comment by adamretter** [comment]
@dizzzz @reinhapa I kept the journal improvements and CI improvements in a single PR, as that is what I had already done for 4.x.x, 5.x.x, and 6.x.x; albeit in those cases I needed to do both in a single PR to be able to see what was going on. If you feel this PR here must be split into two - let me know and I will do so...

File: .github/workflows/sonarcloud.yml (hunk 2 above)
Comment (adamretter) [reply .github/workflows/sonarcloud.yml]: @dizzzz I have managed to move the version into `exist-parent/pom.xml` now, which means it should be updated by dependabot etc.

**Comment by dizzzz** [comment]
It is OK for me, as it is intentional

**Review by dizzzz — APPROVED** [review]

**Comment by duncdrum** [comment]
I m also curious what the JDK has to do with it. Since we use temurin not just on CI here, but also in Docker and all other repos in the org, I m not a fan of changing it without reason. If there is a reason to change, we should stay consistent elsewhere in the org.

**Comment by reinhapa** [comment]
I'm ok changing the JDK for the sake of having the old versions, but on the `develop` branch I would prefer to leave it using the temurin one...

**Comment by dizzzz** [comment]
It looks like that switching to the JVM Adam suggested all kind of strange test/fork/... issues occur.

Looking at the discussions it is probably better to split the jvm change and the recovery log code delta.

**Comment by reinhapa** [comment]
> It looks like that switching to the JVM Adam suggested all kind of strange test/fork/... issues occur.
> 
> Looking at the discussions it is probably better to split the jvm change and the recovery log code delta.

We then at least can rule out side effects from the JDK change... Just to be clear I'm full in the hiding suff!

**Review by line-o — CHANGES_REQUESTED** [review]
I do appreciate hiding the journal progress bar in CI!
I would like to see as @dizzz and @reinhapa  that only the changes necessary for that to work would be included in this PR.
I would also like to see them in just one or two commits.

**Comment by adamretter** [comment]
> I m also curious what the JDK has to do with it.

@duncdrum (cc @reinhapa) As I wrote before you posted, see: "as that is what I had already done for 4.x.x, 5.x.x, and 6.x.x"

> Since we use temurin not just on CI here, but also in Docker and all other repos in the org, I m not a fan of changing it without reason.

@duncdrum We are now using Liberica (and not Temurin) for eXist-db [4.x.x](https://github.com/eXist-db/exist/pull/5468), [5.x.x](https://github.com/eXist-db/exist/pull/5467), and [6.x.x](https://github.com/eXist-db/exist/pull/5466). I am also a fan of consistency and so I wanted to use Liberica also for 7.x.x in THIS repo.
Regards Docker, I think you may be mistaken, we are not using Temurin and we never have:

1. For eXist-db 7.x.x in Docker we are using OpenJDK (from `gcr.io/distroless/java17:latest` base image), see: https://github.com/GoogleContainerTools/distroless/blob/main/java/README.md
2. For eXist-db 6.x.x in Docker we are using OpenJDK, see: https://github.com/eXist-db/exist/blob/develop-6.x.x/exist-docker/src/main/resources-filtered/Dockerfile#L28
3. For eXist-db 5.x.x in Docker we are using OpenJDK, see: https://github.com/eXist-db/exist/blob/develop-5.x.x/exist-docker/src/main/resources-filtered/Dockerfile#L28
4. For eXist-db 4.x.x in Docker we are using OpenJDK, see: https://github.com/eXist-db/docker-existdb/blob/master/Dockerfile#L28

>  If there is a reason to change

Yes, the features of eXist-db 4.x.x, 5.x.x, and 6.x.x only reliably run 100% on Java 8. Whilst you can run some parts of eXist-db 4.x.x, 5.x.x, and 6.x.x on newer Java, you will find various parts that throw exceptions due to changes in the JDK after Java 8; JAXB, Mail, Activation, Image, etc.

There does not appear to be a Temurin build available for Java 8 in macOS (at least in CI) so we simply can't use Temurin in CI, however there is one for Liberica. That is the reason for the change.

I see a reason to drop Temurin, but no reason not to use Liberica (which is also based on OpenJDK).

> we should stay consistent elsewhere in the org.

I am not so concerned about other repos in this org as they each have different requirements that I expect to be set by the owners of those repos.

As the JDK change seems to be of concern, I will move it to a separate PR. However I think that we should choose one JDK vendor across 4.x.x, 5.x.x, 6.x.x, and 7.x.x and stick to it in this repo. As Temurin won't work for us (no JDK 8 support on macOS in CI) we need to choose one of the others.

File: .github/workflows/ci-deploy.yml (hunk 3 above)
Comment (adamretter) [reply .github/workflows/ci-deploy.yml]: Moved JDK change to PR - https://github.com/eXist-db/exist/pull/5480

File: .github/workflows/ci-test.yml (hunk 5 above)
Comment (adamretter) [reply .github/workflows/ci-test.yml]: https://github.com/eXist-db/exist/pull/5480

File: .github/workflows/ci-test.yml (hunk 4 above)
Comment (adamretter) [reply .github/workflows/ci-test.yml]: Moved JDK change to PR - https://github.com/eXist-db/exist/pull/5480

File: .github/workflows/ci-xqts.yml (hunk 7 above)
Comment (adamretter) [reply .github/workflows/ci-xqts.yml]: Moved JDK change to PR - https://github.com/eXist-db/exist/pull/5480

File: .github/workflows/ci-test.yml (hunk 6 above)
Comment (adamretter) [reply .github/workflows/ci-test.yml]: Moved JDK change to PR - https://github.com/eXist-db/exist/pull/5480

File: .github/workflows/sonarcloud.yml (hunk 8 above)
Comment (adamretter) [reply .github/workflows/sonarcloud.yml]: Moved JDK change to PR - https://github.com/eXist-db/exist/pull/5480

**Comment by adamretter** [comment]
@dizzzz @reinhapa I have moved JDK change out of this PR and into a new PR - https://github.com/eXist-db/exist/pull/5480

**Pull request merged**